### ISSUE TITLE
Build system incorrectly chooses intel compiler executable despite -compiler gnu option set

### ISSUE THREAD

**Issue by platipodium** [noaa-ocs-modeling/ADC-WW3-NWM-NEMS#51]
When choosing a `gnu` compiler, the build system chooses mpiifort (`intel`). 

```
$> bash ./build.sh -plat macports -component SCHISM -compiler gnu
```

```
...
mpiifort -g -qopenmp -mkl=sequential -align array32byte -lmkl_intel_lp64 -lmkl_core -lmkl_sequential -lpthread -qopenmp -convert big_endian -assume byterecl -mkl=sequential  -I/opt/local/include/esmf -I/opt/local/include/esmf -I/opt/local/include -I/opt/local/include -I/incmod/nemsio -I/opt/local/include -fp-model strict -c ENS_CplComp_ESMFMod_STUB.f90
```

and fails with `mpiifort not found`

**Comment by zacharyburnettNOAA** [comment]
@platipodium is this fixed by Takis' changes in d2e0a24?

**Comment by platipodium** [comment]
Fixed by the above commit, thanks.
